dashboard/app/helpers/levels_helper.rb

```diff
@@ -21,6 +21,9 @@ def build_script_level_path(script_level, params = {})
       end
     elsif script_level.stage.lockable?
       script_lockable_stage_script_level_path(script_level.script, script_level.stage, script_level, params)
+    elsif script_level.bonus
+      params[:id] = script_level.id
```

Comment:
this feels a little gross; how hard would it be to implement this in a way that didn't modify params?

Reply:
Not hard at all, done.